src/plugins/es_ui_shared/public/request/use_request.ts

```diff
@@ -53,7 +53,7 @@ export const useRequest = <D = any, E = Error>(
 
   // Consumers can use isInitialRequest to implement a polling UX.
   const requestCountRef = useRef<number>(0);
-  const isInitialRequest = requestCountRef.current === 0;
+  const isInitialRequest = requestCountRef.current <= 1; // isInitialRequest === true if no request are sent or if one is sent
```

Comment:
This causes broken behavior in the Snapshot and Restore app, due to the way it branches conditionally on `isInitialRequest` (https://github.com/elastic/kibana/blob/master/x-pack/plugins/snapshot_restore/public/application/sections/home/restore_list/restore_list.tsx#L72).

I found an alternative solution which avoids breaking Snapshot and Restore, while still passing the tests. Can you try this out and let me know if it works in your app, too?

My suggestion is that we use a ref to store this state, since what we are really trying to communicate to consumers is a state in which either "no requests have resolved yet" or "at least one request has resolved". To implement this, we can change this line to:

```ts
const isInitialRequestRef = useRef<boolean>(true);
```

Then lines 100-106 need to change in order to update this state once a request resolves:

```ts
    // Ignore outdated or irrelevant data.
    if (isOutdatedRequest || isUnmounted) {
      return;
    }

    // Surface to consumers that at least one request has resolved.
    isInitialRequestRef.current = false;
```

And then on line 157 we need to consume the ref in the hook's return value:

```ts
  return {
    isInitialRequest: isInitialRequestRef.current,
    isLoading,
    error,
    data,
    sendRequest, // Gives the user the ability to manually request data
  };
```

Reply:
Just updated my PR with your solution is working well for our use case too 👍